File: backend/app/models/task.rb
class Task < ApplicationRecord
  validates :title, presence: true, length: { minimum: 1, maximum: 255 }
  validates :description, presence: true
  validates :status, presence: true, inclusion: { in: ['todo', 'in_progress', 'done'] }
  before_validation :set_default_status, on: :create
  
  scope :todo, -> { where(status: 'todo') }
  scope :in_progress, -> { where(status: 'in_progress') }
  scope :done, -> { where(status: 'done') }
  
  private
  
  def set_default_status
    self.status ||= 'todo'
  end
end

File: frontend/src/types/Task.ts
export interface Task {
  id: number;
  title: string;
  description: string;
  status: 'todo' | 'in_progress' | 'done';
  created_at: string;
  updated_at: string;
}

export interface TaskInput {
  title: string;
  description: string;
  status?: 'todo' | 'in_progress' | 'done';
}

export interface ApiResponse<T> {
  data?: T;
  errors?: string[];
}

export interface AITaskSuggestion {
  title: string;
  description: string;
}
